### ISSUE TITLE
Error when syntax errors in an untitled/unsaved buffer

### ISSUE THREAD

**Issue by brettcannon** [microsoft/pylance-release#1279]
<h3> Environment data </h3>

-   Language Server version: 2021.5.3-pre.1
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>

Get red squiggles for errors.

<h3> Actual behaviour </h3>

"Problems" picks up on the syntax errors, but no red squiggles.

<h3> Logs </h3>

Python Language Server Log
```
Info  - 10:52:50 a.m.] Pylance language server 2021.5.3-pre.1 (pyright df8b6b3c) starting
[Info  - 10:52:50 a.m.] Server root directory: c:\Users\brcan\.vscode-insiders\extensions\ms-python.vscode-pylance-2021.5.3-pre.1\dist
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received null
    at validateString (internal/validators.js:124:11)
    at Object.basename (path.js:671:5)
    at Object.getFileName (c:\Users\brcan\.vscode-insiders\extensions\ms-python.vscode-pylance-2021.5.3-pre.1\dist\pyright-internal\src\common\pathUtils.ts:535:17)
    at listener (c:\Users\brcan\.vscode-insiders\extensions\ms-python.vscode-pylance-2021.5.3-pre.1\dist\pyright-internal\src\analyzer\service.ts:1273:38)
    at FSWatcher.<anonymous> (c:\Users\brcan\.vscode-insiders\extensions\ms-python.vscode-pylance-2021.5.3-pre.1\dist\pyright-internal\src\languageServerBase.ts:415:21)
    at FSWatcher.emit (events.js:315:20)
    at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:186:12)
```

<h3> Code Snippet / Additional information </h3>

```python
import os
x=
y=
2+3
```

**Comment by savannahostrowski** [comment]
Were you an untitled file in a folder or in single file mode?

**Comment by brettcannon** [comment]
I believe the latter, but I honestly don't remember.

**Comment by savannahostrowski** [comment]
@jakebailey Can we just test both cases to see if the issue can be repro'd?

**Comment by jakebailey** [comment]
I think this was fixed in 2021.8.2.
